src/app/AttributeCache.cpp

```diff
@@ -46,6 +53,37 @@ CHIP_ERROR AttributeCache::UpdateCache(const ConcreteDataAttributePath & aPath,
         handle.RightSize();
 
         state.Set<System::PacketBufferHandle>(std::move(handle));
+
+        //
+        // Clear out the committed data version and only set it again once we have received all data for this cluster.
+        // Otherwise, we may have incomplete data that looks like it's complete since it has a valid data version.
+        //
+        mCache[currentClusterPath.mEndpointId][currentClusterPath.mClusterId].mCommittedDataVersion.ClearValue();
+
+        for (auto path : mRequestPathSet)
+        {
+            if (path.IncludesAllAttributesInCluster(currentClusterPath))
+            {
+                if (!mLastConcreteClusterPath.IsValidConcreteClusterPath() || mLastConcreteClusterPath == currentClusterPath)
+                {
+                    mCache[currentClusterPath.mEndpointId][currentClusterPath.mClusterId].mPendingDataVersion = aPath.mDataVersion;
+                }
+                else
+                {
+                    // commit the version since we moved on from a given cluster to a different cluster
```

Comment:
Shouldn't we do this even when the new cluster is _not_ covered by wildcard-attribute paths?  This seems like it should not be inside this loop at all, or even inside this "have data" block....

Reply:
just update the code and description a little bit. We commit the version when we moved on from a given wildcard attribute path to a different wildcard attributePath when it has data. Assume we have pending version for last wildcard attribute, then after that, we encounter the attribute path that is not intersecting with read request paths with data or attribute path without data, and there is no further wildcard attribute path intersecting with read request paths, it does not matter since we would handle it OnReportEnd

Reply:
My point is this: say we have a mix of wildcard and non-wildcard paths.  Specifically, say we have a wildcard path for cluster A and a single-attribute path for cluster B, with the single attribute being a long list.

Once we have moved on from cluster A to cluster B, why wouldn't we commit at that point, instead of waiting for the next wildcard thing or end-or-report?

That said, now that I am thinking this through it's possible that this whole "commit once we have moved on to a different cluster" thing is in fact assuming server behavior not guaranteed by the spec, and we should be committing on report end after all.  Let's to a followup to get this bit sorted out and make it as specific as possible while not assuming specific server behavior.

Reply:
yep, updated with the way where we committed data version when moved on from cluster A to cluster B
Also track #17286 Sort out when to commit the data version change, thanks